internal/ingress/store/store.go

```diff
@@ -339,7 +354,7 @@ func (s Store) ListGlobalKongClusterPlugins() ([]*configurationv1.KongClusterPlu
 		labels.NewSelector().Add(*req),
 		func(ob interface{}) {
 			p, ok := ob.(*configurationv1.KongClusterPlugin)
-			if ok && s.isValidIngresClass(&p.ObjectMeta) {
+			if ok && s.isValidIngressClass(&p.ObjectMeta, annotations.ExactClassMatch) {
```

Comment:
Please document what the basis of performing exact match (as opposed to `s.ingressClassHandling`) here is.

Reply:
Shouldn't be necessary here--ultimately the justification is because it's what https://docs.google.com/document/d/1qMwurrFiOMoy2EK569y6qY-9cLGKJrkTfloqNyp_J6o/edit#heading=h.js8bwj7x6zou says to do (it goes into the rationale a bit more).

A user-facing doc describing all that (or rather the handling of ingress.class annotations as of 1.0, not so much the historical background also covered in that doc) is planned, but not in scope here. This isn't anything unusual as far as I know, it's just that the old implementation had a lot more ambiguity and/or unintended behavior stemming from its organic evolution over time.

Reuse of `s.ingressClassHandling` for things other than actual Ingresses is something we want to move away from as part of this, with resources instead having a per-resource variable handling mode if we allow users to control it (currently we only plan to allow that for Ingress and KongConsumer).